compute_tools/src/compute.rs

```diff
@@ -958,11 +958,15 @@ impl ComputeNode {
         let mut delay_exit = false;
         let mut state = self.state.lock().unwrap();
         state.terminate_flush_lsn = lsn;
-        if let ComputeStatus::TerminationPending { mode } = state.status {
+        if state.status == ComputeStatus::TerminationPendingFast {
+            state.status = ComputeStatus::Terminated;
+            self.state_changed.notify_all();
+            delay_exit = true
+        } else if state.status == ComputeStatus::TerminationPendingImmediate {
             state.status = ComputeStatus::Terminated;
             self.state_changed.notify_all();
             // we were asked to terminate gracefully, don't exit to avoid restart
-            delay_exit = mode == compute_api::responses::TerminateMode::Fast
+            delay_exit = false
```

Comment:
```suggestion
        if matches!(state.status, ComputeStatus::TerminationPendingFast | ComputeStatus::TerminationPendingImmediate) {
            let delay_exit = state.status == ComputeStatus::TerminationPendingFast;
            self.set_status(ComputeStatus::Terminated, &self.state_changed);
```

Not sure if you can use the `set_status` method, but worth a try.

Reply:
No, we don't want to update COMPUTE_CTL_UP. Also, I don't like the solution with "matches".

Reply:
Why do we want code duplication? Use a regular match statement if that looks better.